<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>853</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Kinematic Design</string>
  </property>
  <widget class="QWidget" name="centralWidget"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport">
     <property name="title">
      <string>Export</string>
     </property>
     <addaction name="actionExportSTL"/>
     <addaction name="actionExportSCAD"/>
    </widget>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionSaveImage"/>
    <addaction name="separator"/>
    <addaction name="menuExport"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuMode">
    <property name="title">
     <string>Mode</string>
    </property>
    <widget class="QMenu" name="menuFixed_Body">
     <property name="title">
      <string>Fixed Body</string>
     </property>
     <addaction name="actionFixedRectangle"/>
     <addaction name="actionFixedCircle"/>
     <addaction name="actionFixedPolygon"/>
    </widget>
    <widget class="QMenu" name="menuMovingBody">
     <property name="title">
      <string>Moving Body</string>
     </property>
     <addaction name="actionMovingRectangle"/>
     <addaction name="actionMovingCircle"/>
     <addaction name="actionMovingPolygon"/>
    </widget>
    <addaction name="actionSelect"/>
    <addaction name="separator"/>
    <addaction name="menuFixed_Body"/>
    <addaction name="menuMovingBody"/>
    <addaction name="separator"/>
    <addaction name="actionLinkageRegion"/>
    <addaction name="actionLinkageAvoidance"/>
    <addaction name="separator"/>
    <addaction name="actionKinematics"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="separator"/>
    <addaction name="actionDelete"/>
    <addaction name="actionSelectAll"/>
   </widget>
   <widget class="QMenu" name="menuLayer">
    <property name="title">
     <string>Layer</string>
    </property>
    <addaction name="actionAddLayer"/>
    <addaction name="actionInsertLayer"/>
    <addaction name="actionDeleteLayer"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuKinematics">
    <property name="title">
     <string>Kinematics</string>
    </property>
    <addaction name="actionGenerateLinkage"/>
    <addaction name="separator"/>
    <addaction name="actionRun"/>
    <addaction name="actionRunBackward"/>
    <addaction name="actionStop"/>
    <addaction name="separator"/>
    <addaction name="actionStepForward"/>
    <addaction name="actionStepBackward"/>
    <addaction name="separator"/>
    <addaction name="actionCollisionCheck"/>
    <addaction name="actionRestrictMotionRange"/>
    <addaction name="actionOptions"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionShowSolutions"/>
    <addaction name="actionShowGridLines"/>
    <addaction name="actionShowInputPoses"/>
    <addaction name="separator"/>
    <addaction name="actionRenderBasic"/>
    <addaction name="actionRenderSSAO"/>
    <addaction name="actionRenderLine"/>
    <addaction name="actionRenderHatching"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuMode"/>
   <addaction name="menuLayer"/>
   <addaction name="menuKinematics"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionSelect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>Resources/select.png</normaloff>Resources/select.png</iconset>
   </property>
   <property name="text">
    <string>Select</string>
   </property>
  </action>
  <action name="actionFixedRectangle">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>Resources/fixed_rectangle.png</normaloff>Resources/fixed_rectangle.png</iconset>
   </property>
   <property name="text">
    <string>Rectangle</string>
   </property>
  </action>
  <action name="actionFixedCircle">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>Resources/fixed_circle.png</normaloff>Resources/fixed_circle.png</iconset>
   </property>
   <property name="text">
    <string>Circle</string>
   </property>
  </action>
  <action name="actionFixedPolygon">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>Resources/fixed_polygon.png</normaloff>Resources/fixed_polygon.png</iconset>
   </property>
   <property name="text">
    <string>Polygon</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset>
     <normaloff>Resources/new.png</normaloff>Resources/new.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset>
     <normaloff>Resources/copy.png</normaloff>Resources/copy.png</iconset>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionSelectAll">
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset>
     <normaloff>Resources/delete.png</normaloff>Resources/delete.png</iconset>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset>
     <normaloff>Resources/paste.png</normaloff>Resources/paste.png</iconset>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset>
     <normaloff>Resources/undo.png</normaloff>Resources/undo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset>
     <normaloff>Resources/redo.png</normaloff>Resources/redo.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset>
     <normaloff>Resources/open.png</normaloff>Resources/open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset>
     <normaloff>Resources/save.png</normaloff>Resources/save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionAddLayer">
   <property name="text">
    <string>Add Layer</string>
   </property>
  </action>
  <action name="actionInsertLayer">
   <property name="text">
    <string>Insert Layer</string>
   </property>
  </action>
  <action name="actionDeleteLayer">
   <property name="text">
    <string>Delete Layer</string>
   </property>
  </action>
  <action name="actionGenerateLinkage">
   <property name="icon">
    <iconset>
     <normaloff>Resources/fourbar_linkage.png</normaloff>Resources/fourbar_linkage.png</iconset>
   </property>
   <property name="text">
    <string>Generate Linkage</string>
   </property>
  </action>
  <action name="actionCollisionCheck">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Collision Check</string>
   </property>
  </action>
  <action name="actionLinkageRegion">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>Resources/linkage_region.png</normaloff>Resources/linkage_region.png</iconset>
   </property>
   <property name="text">
    <string>Linkage Region</string>
   </property>
  </action>
  <action name="actionKinematics">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Kinematics</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="icon">
    <iconset>
     <normaloff>Resources/run.png</normaloff>Resources/run.png</iconset>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
  </action>
  <action name="actionRunBackward">
   <property name="text">
    <string>Run Backward</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="icon">
    <iconset>
     <normaloff>Resources/stop.png</normaloff>Resources/stop.png</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
  <action name="actionStepForward">
   <property name="icon">
    <iconset>
     <normaloff>Resources/step_forward.png</normaloff>Resources/step_forward.png</iconset>
   </property>
   <property name="text">
    <string>Step Forward</string>
   </property>
   <property name="shortcut">
    <string>Right</string>
   </property>
  </action>
  <action name="actionStepBackward">
   <property name="icon">
    <iconset>
     <normaloff>Resources/step_backward.png</normaloff>Resources/step_backward.png</iconset>
   </property>
   <property name="text">
    <string>Step Backward</string>
   </property>
   <property name="shortcut">
    <string>Left</string>
   </property>
  </action>
  <action name="actionGenerateSliderCrank">
   <property name="icon">
    <iconset>
     <normaloff>Resources/slider_crank.png</normaloff>Resources/slider_crank.png</iconset>
   </property>
   <property name="text">
    <string>Generate Slider Crank</string>
   </property>
  </action>
  <action name="actionShowSolutions">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Solutions</string>
   </property>
  </action>
  <action name="actionRenderBasic">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Basic</string>
   </property>
  </action>
  <action name="actionRenderSSAO">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>SSAO</string>
   </property>
  </action>
  <action name="actionRenderLine">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Line</string>
   </property>
  </action>
  <action name="actionRenderHatching">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Hatching</string>
   </property>
  </action>
  <action name="actionSaveSTL">
   <property name="text">
    <string>Save STL</string>
   </property>
  </action>
  <action name="actionExportSTL">
   <property name="text">
    <string>STL</string>
   </property>
  </action>
  <action name="actionExportSCAD">
   <property name="text">
    <string>SCAD</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="icon">
    <iconset>
     <normaloff>Resources/options.png</normaloff>Resources/options.png</iconset>
   </property>
   <property name="text">
    <string>Options</string>
   </property>
  </action>
  <action name="actionMovingRectangle">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>Resources/moving_rectangle.png</normaloff>Resources/moving_rectangle.png</iconset>
   </property>
   <property name="text">
    <string>Rectangle</string>
   </property>
  </action>
  <action name="actionMovingCircle">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>Resources/moving_circle.png</normaloff>Resources/moving_circle.png</iconset>
   </property>
   <property name="text">
    <string>Circle</string>
   </property>
  </action>
  <action name="actionMovingPolygon">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>Resources/moving_polygon.png</normaloff>Resources/moving_polygon.png</iconset>
   </property>
   <property name="text">
    <string>Polygon</string>
   </property>
  </action>
  <action name="actionFixedRectangle_2">
   <property name="text">
    <string>Rectangle</string>
   </property>
  </action>
  <action name="actionAho">
   <property name="text">
    <string>Aho</string>
   </property>
  </action>
  <action name="actionLinkageAvoidance">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>Resources/linkage_avoidance.png</normaloff>Resources/linkage_avoidance.png</iconset>
   </property>
   <property name="text">
    <string>Linkage Avoidance Region</string>
   </property>
  </action>
  <action name="actionGenerateWattI">
   <property name="icon">
    <iconset>
     <normaloff>Resources/watt_i_linkage.png</normaloff>Resources/watt_i_linkage.png</iconset>
   </property>
   <property name="text">
    <string>Generate Watt I</string>
   </property>
  </action>
  <action name="actionShowGridLines">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Grid Lines</string>
   </property>
  </action>
  <action name="actionShowInputPoses">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Input Poses</string>
   </property>
  </action>
  <action name="actionSaveImage">
   <property name="text">
    <string>Save Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionRestrictMotionRange">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Restrict Motion Range</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="MainWindow.qrc"/>
 </resources>
 <connections/>
</ui>
